src/pages/givingform.tsx

```diff
@@ -0,0 +1,834 @@
+import React, { useState, useEffect } from "react"
+import { useStaticQuery, graphql } from "gatsby"
+import { useForm } from "react-hook-form"
+import classNames from "classnames"
+
+import Layout from "../components/layout"
+import formStyles from "../components/form.module.scss"
+
+const googleFormId = "1FAIpQLSeED6ffZVCYFG5amGWUtTGWkr8EI-rbPO2i_f-z0Ur6XvTNTw"
+
+const developmentWarning = (
+    <div>
+        <em>
+            Warning - this form currently submits to a CCM GSuite Google
+            Spreadsheet owned by{" "}
+            <a href="mailto:tom@christchurchmayfair.org">
+                tom@christchurchmayfair.org
+            </a>{" "}
+            and is only for test purposes. Please do not submit any data that
+            you do not wish him to see!
+        </em>
+    </div>
+)
+
+const fieldNameToEntryId: { [fieldName: string]: string } = {
+    title: "1412836924",
+    givenName: "403087895",
+    familyName: "383939029",
+    streetAddress: "2131927972",
+    extraAddressLine: "54229700",
+    townCity: "1247583245",
+    postCode: "524386314",
+    telephoneNumber: "899696842",
+    emailAddress: "1518406431",
+    giftAmountInGBP: "710981495",
+    giftType: "1615869123",
+    regularGiftFrequency: "2138441357",
+    regularGiftCommencementDate: "825676099",
+    thisGiftIsEligibleForGiftAid: "567316447",
+    allFutureGiftsAreEligibleForGiftAid: "518456171",
+    retrospectivelyReclaimGiftAidForFourYears: "1656623006",
+}
+
+type GiftFormData = {
+    title: string
+    givenName: string
+    familyName: string
+    streetAddress: string
+    extraAddressLine?: string
```

Comment:
Warning – personal opinion up ahead.

For structs like this, my feeling is that rather than doing
```ts
someOptionalField?: string
```

It's better to do
```ts
someOptionalField: string | null,
```

The problem with the first is that it lets you create that struct without specifying that value at all, whereas the second forces you to be explicit (and my general philosophy is that explicit is better than implicit).

Obviously there *are* times when the first example would make a better API (eg, I often use it for optional react props). But (in my opinion) this isn't one of them.

Reply:
Makes perfect sense.